nvmf-autoconnect/dracut-95nbft/README.md

```diff
@@ -0,0 +1,125 @@
+# The dracut 95nbft module
+
+Focused solely on providing the Boot from NVMe over TCP functionality, intended
+to replace parts of the existing `95nvmf` dracut module. At the moment this all
+depends on the recently added NetworkManager NBFT support, though the desire is
+to support more network management frameworks in the future.
+
+Related nvme-cli meson configure options:
+* `-Ddracut-module` (default=false) - enables the 95nbft dracut module
+* `-Ddracutmodulesdir` (default=`$prefix/lib/dracut/modules.d/`)
+* `-Dnetworkmanagerdir` (default=`$prefix/lib/NetworkManager/`)
+
+
+# The design
+
+(see [dracut.bootup(7)](https://man7.org/linux/man-pages/man7/dracut.bootup.7.html)
+for the overall boot process flow)
+
+The boot process looks roughly as follows:
+* `nbft-boot-pre.service` is run, creates udev network link files and tells
+  dracut to activate networking
+* dracut runs `nm-initrd-generator` and starts the NetworkManager daemon
+* `systemd-udev-trigger.service` renames the network interfaces
+* `nm-wait-online-initrd.service` finishes, indicating networking is up and ready
+* `nbft-boot-connect.service` initiates actual NVMe connections
+* the dracut initqueue is waiting for specific block devices (rootfs) to appear
+
```

Comment:
So this, and the starting of NM above, are the only 2 things that dracut needs to perform, and therefore it's relatively easy to plug this scheme into mkosi or another initramfs generator, as long as that generator is based on systemd. Nice.

Am I understanding correctly?